packages/web-app-files/src/components/SideBar/Shares/Links/CreateQuickLink.vue

```diff
@@ -0,0 +1,65 @@
+<template>
+  <div class="oc-mb-s oc-width-1-1 oc-mb-l">
+    <h4 class="oc-text-truncate oc-files-file-link-name oc-my-s" v-text="heading" />
+    <div class="oc-flex oc-flex-middle oc-flex-between oc-width-1-1 oc-p-s link-name-container">
+      <div class="oc-flex oc-flex-middle oc-text-truncate">
+        <oc-icon name="link" fill-type="line" />
+        <p class="oc-files-file-link-url oc-ml-s oc-text-truncate oc-my-rm" v-text="noLinkLabel" />
+      </div>
+      <oc-button
+        v-oc-tooltip="createLinkHint"
+        appearance="raw"
+        :aria-label="createLinkHint"
+        @click="createQuickLink"
+        v-text="createLinkLabel"
+      />
+    </div>
```

Comment:
Just a note: UX-wise I would've expected to see the role selection already in the create form. No need to do that in this PR. But let's discuss that as a possible improvement in a couple of days.

Reply:
Yes, there's also screendesigns replacing the whole "createForm" for normal public links with what looks like the solution you're suggesting. But this in combination with enforced expiration dates/enforced password for certain roles always breaks my brain

Reply:
certainly nothing for now 😃

Reply:
Issue is party tracked in https://github.com/owncloud/web/issues/6782 already